code/app/src/main/java/com/example/bread/repository/ParticipantRepository.java

```diff
@@ -216,4 +217,23 @@ public void checkIfAlreadyFollowed(@NonNull String username, String followerUser
 
     public void checkIfAlreadyFollowing(@NonNull String username, String followingUsername, @NonNull OnSuccessListener<Boolean> onSuccessListener, OnFailureListener onFailureListener) {
     }
+
+    public void sendFollowRequest(@NonNull String fromUsername, @NonNull String toUsername,
+                                  @NonNull OnSuccessListener<Void> onSuccessListener,
+                                  OnFailureListener onFailureListener) {
+        Map<String, Object> request = new HashMap<>();
+        request.put("fromUsername", fromUsername);
+        request.put("status", "pending");
+        request.put("timestamp", com.google.firebase.Timestamp.now());
```

Comment:
This method will potentially change if you based it of farhan's PR. I would recommend leaving a todo with comments on how and where to integerate `NotificationService.sendFollowRequestNotification(fromUsername, toUsername);`